### ISSUE TITLE
Rename mutate to define

### ISSUE THREAD

**Issue by dodger487** [dodger487/dplython#46]
We have already differed from dplyr by using `sift` instead of `filter`. `mutate` seems like a misnomer-- we're not mutating an existing column, but rather we're adding a new column. We're changing the dataframe that is output, but we're doing that with all of the verbs.

To make things a little easier to understand for new users, we could rename `mutate` to `define`. We could also keep around `mutate` for the hardcore dplyr users.

**Comment by danrobinson** [comment]
+1. Curious to hear thoughts from @dgrtwo

**Comment by ngould** [comment]
I don't love mutate() either, but "define" is a crowded namespace in the land of software concepts. How about annex()?

**Comment by danrobinson** [comment]
I disagree. To the extent `define` has a common meaning, it's exactly the same meaning as is used here. The user is defining a new variable in terms of previously existing variables.

**Comment by csaid** [comment]
- 1 to define(). I think annex() is a bit too obscure.

**Comment by ngould** [comment]
I can get on board with that rationale, @danrobinson.
